### DISCUSSION TITLE
How to handle highlighting for embedding languages?

### DISCUSSION THREAD

**Discussion by winston0410** [tree-sitter/tree-sitter#1307] [Q&A] [answered]
Is it possible to provide highlight to an embedded languages, like a code block in Markdown? I think to handle this issue, the parser needs to recognize the filetype, and to be able to pull the parser logic of that specific embedded language. Has someone attempted something like this before?

**Comment by razzeee** [comment, marked as answer]
There are these docs for highlighting https://tree-sitter.github.io/tree-sitter/3-syntax-highlighting.html#language-injection
